benchmarks/nemo/nemolite2d/kernels/c_family/boundary_conditions_kern.c

```diff
@@ -1,174 +1,180 @@
+    ret = clSetKernelArg(bc_flather_u, arg_idx++, sizeof(cl_mem),
+		       (void *)hu_device);
+    check_status("clSetKernelArg", ret);
+    ret = clSetKernelArg(bc_flather_u, arg_idx++, sizeof(cl_mem),
+		       (void *)sshn_u_device);
+    check_status("clSetKernelArg", ret);
+    ret = clSetKernelArg(bc_flather_u, arg_idx++, sizeof(cl_mem),
+		       (void *)tmask_device);
+    check_status("clSetKernelArg", ret);
+    ret = clSetKernelArg(bc_flather_u, arg_idx++, sizeof(cl_double),
+		       (void *)g);
+    check_status("clSetKernelArg", ret);
+
+    fprintf(stdout, "Set %d arguments for bc_flather_u kernel\n", arg_idx);
 
-/*
-  subroutine invoke_bc_ssh(istep, ssha)
-    implicit none
-    integer,            intent(in)    :: istep
-    type(r2d_field),    intent(inout) :: ssha
-    ! Locals
-    integer  :: ji, jj
+}
 
-    DO jj = ssha%internal%ystart, ssha%internal%ystop
-       DO ji = ssha%internal%xstart, ssha%internal%xstop
-          call bc_ssh_code(ji, jj, &
-                           istep, ssha%data, ssha%grid%tmask)
-       END DO
-    END DO
+/* Set OpenCL Kernel Parameters for bc_flather_v kernel */
+void set_args_bc_flather_v(cl_kernel bc_flather_v,
+			 cl_int *width,
+			 cl_int *xstop,
+			 cl_mem *va_device,
+			 cl_mem *hv_device,
+			 cl_mem *sshn_v_device,
+			 cl_mem *tmask_device,
+             cl_double *g){
+    cl_int ret;
+    int arg_idx = 0;
+    ret = clSetKernelArg(bc_flather_v, arg_idx++, sizeof(cl_int),
+		       (void *)width);
+    check_status("clSetKernelArg", ret);
+    ret = clSetKernelArg(bc_flather_v, arg_idx++, sizeof(cl_int),
+		       (void *)xstop);
+    check_status("clSetKernelArg", ret);
+    ret = clSetKernelArg(bc_flather_v, arg_idx++, sizeof(cl_mem),
+		       (void *)va_device);
+    check_status("clSetKernelArg", ret);
+    ret = clSetKernelArg(bc_flather_v, arg_idx++, sizeof(cl_mem),
+		       (void *)hv_device);
+    check_status("clSetKernelArg", ret);
+    ret = clSetKernelArg(bc_flather_v, arg_idx++, sizeof(cl_mem),
+		       (void *)sshn_v_device);
+    check_status("clSetKernelArg", ret);
+    ret = clSetKernelArg(bc_flather_v, arg_idx++, sizeof(cl_mem),
+		       (void *)tmask_device);
+    check_status("clSetKernelArg", ret);
+    ret = clSetKernelArg(bc_flather_v, arg_idx++, sizeof(cl_double),
+		       (void *)g);
+    check_status("clSetKernelArg", ret);
+
+    fprintf(stdout, "Set %d arguments for bc_flather_v kernel\n", arg_idx);
+  
+}
 
-  end subroutine invoke_bc_ssh
-*/
+#endif
+#endif
 
 #ifdef __OPENCL_VERSION__
-__kernel void bc_ssh_code(int width,
+__kernel void bc_ssh_code(int width, int xstop,
               int istep,
               __global double* restrict ssha,
               __global int* restrict tmask,
               double rdt){
   int ji = get_global_id(0);
   int jj = get_global_id(1);
-  int nrow = (int)get_global_size(1);
-  if(ji==0 || ji > (width-2))return;
-  if(jj==0 || jj > (nrow-2))return;
+  if(ji > xstop)return;
```

Comment:
Yes, the offset and global_size of the NDRange are enough to delimit `jj` for now (in FPGAs a square 2D local size may be worth it but since I haven't got good performance with it yet I am not putting it into master yet)